scripts/jenkins/build

```diff
@@ -27,10 +27,15 @@
 set -a
 
 # exit with error code on any command failure (including pipes)
-set -e
-set -o pipefail
+set -eo pipefail
+
+if [[ $@ =~ " lnet"( |$) ]]
+then
+    TRANSPORT="lnet"
+else
+    TRANSPORT="libfabric"
+fi
```

Comment:
It's better to specify the transport by an explicit parameter. With this change we can only support 2 transports: lnet and libfabric. This is already incorrect: there are also sock and membulk transports, and more transports can be added in the future.

Reply:
Done. Now build script will accept parameter for transport. However I will add support sock and membulk in another ticket as testing that would increase the scope of this ticket. Let me take that in another ticket.

Reply:
As per the discussion in our scrum, I have added the support for sock and membulk in the same ticket.